src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemQueryTagService.cs

```diff
@@ -53,9 +58,19 @@ public async Task<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> GetQueryTagsA
             return await _queryTagCache.GetAsync(cancellationToken: cancellationToken);
         }
 
-        private async Task<IReadOnlyCollection<WorkitemQueryTagStoreEntry>> ResolveQueryTagsAsync(CancellationToken cancellationToken)
+        private async Task<IReadOnlyCollection<QueryTag>> ResolveQueryTagsAsync(CancellationToken cancellationToken)
         {
-            return await _indexWorkitemStore.GetWorkitemQueryTagsAsync(cancellationToken);
+            var workitemQueryTags = await _indexWorkitemStore.GetWorkitemQueryTagsAsync(cancellationToken);
+
+            foreach (var tag in workitemQueryTags)
+            {
+                if (_dicomTagParser.TryParseToDicomItem(tag.Path, out var dicomItem))
+                {
+                    tag.Item = dicomItem;
+                }
```

Comment:
Log if something falses to parse

Reply:
Logging and throwing an exception, as this should never happen. It can only happen if some one manually change the db